extension/bridge/src/main.rs

```diff
@@ -0,0 +1,128 @@
+        match TcpStream::connect(SERVER_AND_PORT.parse().unwrap()) {
+            Ok(mut stream) => {
+                poll.registry()
+                    .register(&mut stream, VPN, Interest::READABLE | Interest::WRITABLE)
+                    .unwrap();
+                // This second loop processes messages coming from the tcp stream and from the
+                // STDIN thread via the waker/sender.
+                'connected: loop {
+                    poll.poll(&mut events, None).unwrap();
+
+                    for event in events.iter() {
+                        match event.token() {
+                            VPN => {
+                                if event.is_error()
+                                    || event.is_read_closed()
+                                    || event.is_write_closed()
+                                {
+                                    eprintln!("VPN connection dropped");
+                                    poll.registry().deregister(&mut stream).unwrap();
+                                    vpn_connected = false;
+                                    write_vpn_down(false);
+                                    thread::sleep(time::Duration::from_millis(500));
+                                    break 'connected;
+                                }
+
+                                if !vpn_connected && (event.is_readable() || event.is_writable()) {
+                                    vpn_connected = true;
+                                    write_vpn_up();
+                                }
+
+                                if event.is_readable() {
+                                    loop {
+                                        match read_input(&mut stream) {
+                                            Ok(value) => {
+                                                write_output(std::io::stdout(), &value)
+                                                    .expect("Unable to write to STDOUT");
+                                            }
+                                            _ => {
+                                                break;
+                                            }
+                                        }
+                                    }
+                                }
+                            }
+                            STDIN_WAKER => {
+                                let value = receiver.recv().unwrap();
+                                match write_output(&mut stream, &value) {
+                                    Ok(_) => {}
+                                    _ => {
+                                        write_vpn_down(true);
+                                    }
+                                }
+                            }
+                            _ => unreachable!(),
+                        }
+                    }
+                }
+            }
+            _ => {
+                eprintln!("No connection to the VPN client");
```

Comment:
Wouldn't this get kind of spammy every time the user loads their browser without the VPN client running?